src/StorageSync/StorageSync/SyncGroup/NewSyncGroupCommand.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.StorageSync.Common;
+using Microsoft.Azure.Commands.StorageSync.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.StorageSync.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.StorageSync;
+using Microsoft.Azure.Management.StorageSync.Models;
+using System.Collections;
+using System.Management.Automation;
+using StorageSyncModels = Microsoft.Azure.Management.StorageSync.Models;
+
+namespace Microsoft.Azure.Commands.StorageSync.SyncGroup
+{
+
+    [Cmdlet(VerbsCommon.New, StorageSyncNouns.NounAzureRmStorageSyncGroup, DefaultParameterSetName = StorageSyncParameterSets.ObjectParameterSet), OutputType(typeof(PSSyncGroup))]
```

Comment:
Must implement ShouldProcess pattern, and have SupportsShouldProcess=true here: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/should-process-confirm-impact.md

Reply:
Implement ShouldProcess on New* cmdlets

---
In reply to: [255700492](https://github.com/Azure/azure-powershell/pull/8465#discussion_r255700492) [](ancestors = 255700492)